src/pages/home/ReportScreen.js

```diff
@@ -147,9 +152,14 @@ class ReportScreen extends React.Component {
      * @returns {Boolean}
      */
     shouldShowLoader() {
+        const reportIDFromPath = getReportID(this.props.route);
+
         // This means there are no reportActions at all to display, but it is still in the process of loading the next set of actions.
         const isLoadingInitialReportActions = _.isEmpty(this.props.reportActions) && this.props.report.isLoadingReportActions;
-        return !getReportID(this.props.route) || isLoadingInitialReportActions || !this.props.report.reportID;
+
+        // This is necessary so that when we are retrieving the next report data from Onyx the ReportActionsView will remount completely
+        const isTransitioning = this.props.report && this.props.report.reportID !== reportIDFromPath;
```

Comment:
@marcaaron I know it's been a long time, but any chance you can help explain how the `reportID` from route params could be different than the `reportID` from the `report` that came from `withOnyx`? Since the key passed to `withOnyx` uses the `reportID` from the `route` param, I'd think that they could never be different. Trying to wrap my head around this

Reply:
😅 

I want to say something like:

- the route changes 
- triggers a re-render
- start fetching the report from Onyx
- the report takes a long time to return
- during that time this component will be "transitioning"

Reply:
I think that explanation makes sense because if the props change then `withOnyx` callbacks won't necessarily return the report right away, but I'd have to do some debugging to see if it's still possible with all the changes since then.